test/e2e/reboot.go

```diff
@@ -147,26 +196,9 @@ func rebootNode(c *client.Client, provider, name string, result chan bool) {
 	}
 
 	// Reboot the node.
-	Logf("Getting external IP address for %s", name)
-	host := ""
-	for _, a := range node.Status.Addresses {
-		if a.Type == api.NodeExternalIP {
-			host = a.Address + ":22"
-			break
-		}
-	}
-	if host == "" {
-		Logf("Couldn't find external IP address for node %s", name)
-		result <- false
-		return
-	}
-	Logf("Calling reboot on %s", name)
-	rebootCmd := "sudo reboot"
-	if _, _, code, err := SSH(rebootCmd, host, provider); code != 0 || err != nil {
-		Failf("Expected 0 exit code and nil error when running %s on %s, got %d and %v",
-			rebootCmd, node, code, err)
-		result <- false
-		return
+	if err = issueSSHCommand(node, provider, rebootCmd); err != nil {
+		// Just log the error as reboot may cause unclean termination of ssh session, which is expected.
```

Comment:
No, I think that you need to error out and fail the test if the ssh session fails. Use nohup or a cron job so that you can cleanly terminate the ssh session before the actual reboot occurs.